### PR TITLE
Changed the units on grid snapping to be relative to the squares in the graph paper....

### PR DESCRIPTION
**Pull request by frohro**
Hi,
Thanks to Fabian, we have a beautiful new way to select the grid size.  I wasn't quite happy with it though, because it was in basic units, and I wanted it in units relative to the size of the squares on the graph paper.  Thanks for a very useful program!
Rob

### CODE REVIEW

File: src/control/settings/Settings.cpp [hunk 1]
```diff
@@ -1104,10 +1104,10 @@ void Settings::setSnapGridTolerance(double tolerance) {
 auto Settings::getSnapGridTolerance() const -> double { return this->snapGridTolerance; }
 auto Settings::getSnapGridSize() const -> double { return this->snapGridSize; };
 void Settings::setSnapGridSize(double gridSize) {
-    if (this->snapGridSize == gridSize) {
+    if (this->snapGridSize == gridSize/14.17) {
         return;
     }
-    this->snapGridSize = gridSize;
+    this->snapGridSize = gridSize*14.17;
```
Comment (Febbe) [inline src/control/settings/Settings.cpp:1110]: Both the units of gridSize and this->snapGridSize should remaint "pt".   
The transformation should be done elsewhere.

**Comment by frohro** [comment]
I understand that in Europe and most of the world, the meter is the unit, and every dimension is based on it.  I love the MKS and SI systems, however, a number of users are in the antiquated USA, where things like graph paper sizes are not always in cm.  Eventually, the user should be able to change the size of the squares on the paper used, and in this case, it would still be desired that  "snap to grid" size would be in units of the size of the grid on the paper.  So I still think the size of the grid is more important than the cm.  As a reference to this, have a look at the [Wikipedia page on graph paper](https://en.wikipedia.org/wiki/Graph_paper).  I'm an engineering professor in the USA, and over here [Engineering paper](https://www.staples-3p.com/s7/is/image/Staples/sp39612981_sc7?wid=512&hei=512) (dark inch subdivisions, with lighter five squares per inch, printed on the back of the paper, so you can still see the grid faintly on the front) is still the standard.  I think the user would want the snap to grid to be the grid that the paper they are using matches, even if it isn't an MKS unit.

**Comment by frohro** [comment]
I like your suggestion of moving the scaling into SettingsDialog.cpp.  I pushed those changes.

**Comment by frohro** [comment]
I'd appreciate some help making the clang format right.  I don't know how.  :-)

**Comment by rolandlo** [comment]
About making the clang format right, let me tell me how I do it.
1) Via command line: execute (in the xournalpp directory) 

`clang-format-9 -i --style=file FILENAME`
where you replace FILENAME with the path to your file which you want to format, e.g. ./src/model/Stroke.cpp

2) If you use visual studio code you can get the desired clang format style applied automatically whenever you save a file. Just make it part of your settings.json file in the .vscode folder. Mine looks like this:
```
{
    "C_Cpp.clang_format_path": "/usr/bin/clang-format-9",
    "C_Cpp.clang_format_fallbackStyle": "none",
    "editor.formatOnSave": true,
    "files.associations": {
        "*.tcc": "cpp",
        "random": "cpp"
    }
}
```
probably other code editors / IDEs have similar options.

**Comment by Febbe** [comment]
To make the clang format right I recommend not to use clang-format directly, because this changes every line where the format was wrong.
Try to use `git-clang-format HEAD~<the number of commits you made>`

File: src/gui/dialog/SettingsDialog.cpp [hunk 2]
```diff
@@ -575,7 +575,7 @@ void SettingsDialog::save() {
             static_cast<double>(gtk_spin_button_get_value(GTK_SPIN_BUTTON(get("spSnapRotationTolerance")))));
     settings->setSnapGridTolerance(
             static_cast<double>(gtk_spin_button_get_value(GTK_SPIN_BUTTON(get("spSnapGridTolerance")))));
-    settings->setSnapGridSize(static_cast<double>(gtk_spin_button_get_value(GTK_SPIN_BUTTON(get("spSnapGridSize")))));
+    settings->setSnapGridSize(static_cast<double>(gtk_spin_button_get_value(GTK_SPIN_BUTTON(get("spSnapGridSize")))*14.17));
```
[the remark this answers is not in the document]
Comment (frohro) [reply src/gui/dialog/SettingsDialog.cpp]: Bravo!  That is what I was looking for!  Why don't we use that on the other 14.17's throughout the code?

**Comment by Febbe** [comment]
> I understand that in Europe and most of the world, the meter is the unit, and every dimension is based on it. I love the MKS and SI systems, however, a number of users are in the antiquated USA, where things like graph paper sizes are not always in cm. Eventually, the user should be able to change the size of the squares on the paper used, and in this case, it would still be desired that "snap to grid" size would be in units of the size of the grid on the paper. 

Yes I fully agree, the snapping should be relative to the grid size. I have an idea in the head, where users can customize the grid and the relative snapping. This should be possible with different units (accepting inch, cm, pt). Also this should be moved to the paper options.

> So I still think the size of the grid is more important than the cm. As a reference to this, have a look at the [Wikipedia page on graph paper](https://en.wikipedia.org/wiki/Graph_paper). I'm an engineering professor in the USA, and over here [Engineering paper](https://www.staples-3p.com/s7/is/image/Staples/sp39612981_sc7?wid=512&hei=512) (dark inch subdivisions, with lighter five squares per inch, printed on the back of the paper, so you can still see the grid faintly on the front) is still the standard. I think the user would want the snap to grid to be the grid that the paper they are using matches, even if it isn't an MKS unit.

For now this is good as you implemented it :).

**Pull request merged**